openedx/core/djangoapps/programs/utils.py

```diff
@@ -582,3 +587,45 @@ def _collect_instructors(self, course_run):
             self.instructors.update(
                 {instructor.get('name'): instructor for instructor in course_instructors.get('instructors', [])}
             )
+
+    def _collect_one_click_purchase_eligibility_data(self):
+        """
+        Extend the program data with data about learner's eligibility for one click purchase,
+        discount data of the program and SKUs of seats that should be added to basket.
+        """
+        applicable_seat_types = self.data['applicable_seat_types']
+        is_learner_eligible_for_one_click_purchase = self.data['is_program_eligible_for_one_click_purchase']
+        skus = []
+        if is_learner_eligible_for_one_click_purchase:
+            for course in self.data['courses']:
+                is_learner_eligible_for_one_click_purchase = not any(
+                    course_run['is_enrolled'] for course_run in course['course_runs']
+                )
+                if is_learner_eligible_for_one_click_purchase:
+                    for seat in course['course_runs'][0]['seats']:
```

Comment:
Unpublished content is indexed by the course catalog? FYI @mjfrey 

Let's take a step back here before we run into problems on production here. How do we determine which SKUs will be sent to the basket page? What's the algorithm for that?

Currently, we're making an assumption I personally dislike. IMHO, course catalog API is responsible for providing necessary information about a program. We currently need information about products that should be added to the basket. If `courses`->`course_runs` is not reliable source of information, another field should be added to the serializer and only the serializer should know the logic behind getting those SKUs.